homeassistant/components/airzone/select.py

```diff
@@ -0,0 +1,158 @@
+
+
+@dataclass
+class AirzoneSelectDescriptionMixin:
+    """Define an entity description mixin for select entities."""
+
+    api_param: str
+    options: dict[str, int]
+
+
+@dataclass
+class AirzoneSelectDescription(SelectEntityDescription, AirzoneSelectDescriptionMixin):
+    """Class to describe an Airzone select entity."""
+
+
+GRILLE_ANGLE_OPTIONS: Final[dict[str, int]] = {
+    "90º": GrilleAngle.DEG_90,
+    "50º": GrilleAngle.DEG_50,
+    "45º": GrilleAngle.DEG_45,
+    "40º": GrilleAngle.DEG_40,
+}
+
+SLEEP_OPTIONS: Final[dict[str, int]] = {
+    "Off": SleepTimeout.SLEEP_OFF,
+    "30m": SleepTimeout.SLEEP_30,
+    "60m": SleepTimeout.SLEEP_60,
+    "90m": SleepTimeout.SLEEP_90,
+}
+
+
+ZONE_SELECT_TYPES: Final[tuple[AirzoneSelectDescription, ...]] = (
+    AirzoneSelectDescription(
+        api_param=API_COLD_ANGLE,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_COLD_ANGLE,
+        name="Cold Angle",
+        options=GRILLE_ANGLE_OPTIONS,
+    ),
+    AirzoneSelectDescription(
+        api_param=API_HEAT_ANGLE,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_HEAT_ANGLE,
+        name="Heat Angle",
+        options=GRILLE_ANGLE_OPTIONS,
+    ),
+    AirzoneSelectDescription(
+        api_param=API_SLEEP,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_SLEEP,
+        name="Sleep",
+        options=SLEEP_OPTIONS,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[AirzoneSelect] = []
+
+    for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in ZONE_SELECT_TYPES:
+            if description.key in zone_data:
+                entities.append(
+                    AirzoneZoneSelect(
+                        coordinator,
+                        description,
+                        entry,
+                        system_zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class AirzoneSelect(AirzoneEntity, SelectEntity):
+    """Define an Airzone select."""
+
+    entity_description: AirzoneSelectDescription
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator updates."""
+        self._async_update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update select attributes."""
+        opt_str = None
+        opt_val = self.get_airzone_value(self.entity_description.key)
+        if opt_val is not None:
+            for key, val in self.entity_description.options.items():
+                if val == opt_val:
+                    opt_str = key
+                    break
```

Comment:
I'd make this a named function so you can return as soon as you find it and it will better show the intent

Reply:
I'm not sure if 1e102f433c0440dd3148e3ff5b2ee113ab3ea9bd is what you are suggesting...